Incorrect error message location

I tried this code:
```rust
// the error is reported here
let em = |v: f64| -> f64 { v };

let x: f64 = em(1i16.into());

// but originates here
let y: f64 = 0.01f64 * 1i16.into();
```
([playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=75e5444eec728a631e6c26a532acce63))

The error message:
```
error[E0284]: type annotations needed for `f64`
 --> src/main.rs:3:15
  |
3 |     let em = |v: f64| -> f64 { v };
  |               ^ consider giving this closure parameter a type
  |
  = note: cannot resolve `<f64 as std::ops::Mul<_>>::Output == f64`
```


I expected to see this happen: The error is reported on the line for `y`.

Instead, this happened: The error is reported for the closure…

### Meta
Nighty, Beta and Stable affected.

--- a/tests/ui/inference/issue-70082.rs
+++ b/tests/ui/inference/issue-70082.rs
@@ -1,0 +1,10 @@
+fn main() {
+    // this closure is fine, and should not get any error annotations
+    let em = |v: f64| -> f64 { v };
+
+    let x: f64 = em(1i16.into());
+
+    let y: f64 = 0.01f64 * 1i16.into();
+    //~^ ERROR type annotations needed
+    //~| HELP try using a fully qualified path
+}

--- a/tests/ui/inference/issue-70082.stderr
+++ b/tests/ui/inference/issue-70082.stderr
@@ -1,0 +1,17 @@
+error[E0284]: type annotations needed
+  --> $DIR/issue-70082.rs:7:33
+   |
+LL |     let y: f64 = 0.01f64 * 1i16.into();
+   |                          -      ^^^^
+   |                          |
+   |                          type must be known at this point
+   |
+   = note: cannot satisfy `<f64 as Mul<_>>::Output == f64`
+help: try using a fully qualified path to specify the expected types
+   |
+LL |     let y: f64 = 0.01f64 * <i16 as Into<T>>::into(1i16);
+   |                            +++++++++++++++++++++++    ~
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0284`.

